### ISSUE TITLE
Cross compilation to aarch64 fails with CMake Error: TRY_RUN() invoked in cross-compiling mode

### ISSUE THREAD

**Issue by paper42** [obsproject/obs-studio#4884]
### Operating System Info

Other

### Other OS

Void Linux

### OBS Studio Version

Other

### OBS Studio Version (Other)

at least 27.0.1 and 27.0.1

### OBS Studio Log URL

\-

### OBS Studio Crash Log URL

\-

### Expected Behavior

cross compilation to aarch64 finishes successfully

### Current Behavior

```
-- Found EGL: /usr/aarch64-linux-gnu/usr/include  
-- Found FFmpeg: /usr/aarch64-linux-gnu/usr/lib/../lib64/libavcodec.so (found version "58.91.100") found components: avcodec avdevice avutil avformat 
-- Found Libcurl: /usr/aarch64-linux-gnu/usr/lib/../lib64/libcurl.so  
-- Scripting: Luajit supported
-- Scripting: Python 3 supported
-- Found SWIG: /usr/bin/swig (found suitable version "4.0.2", minimum required is "2")  
-- Using system Jansson library
-- XCB[XCB]: Found component XCB
-- Found XCB: /usr/aarch64-linux-gnu/usr/lib/libxcb.so  found components: XCB 
-- Found X11_XCB: /usr/aarch64-linux-gnu/usr/lib/libX11-xcb.so  
-- Found Wayland: /usr/aarch64-linux-gnu/usr/lib/libwayland-client.so;/usr/aarch64-linux-gnu/usr/lib/libwayland-server.so;/usr/aarch64-linux-gnu/usr/lib/libwayland-egl.so;/usr/aarch64-linux-gnu/usr/lib/libwayland-cursor.so   
-- Looking for include file pthread.h
-- Looking for include file pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Check if compiler accepts -pthread
CMake Error: TRY_RUN() invoked in cross-compiling mode, please set the following cache variables appropriately:
   THREADS_PTHREAD_ARG (advanced)
For details see /builddir/obs-studio-27.0.1/build/TryRunResults.cmake
-- Check if compiler accepts -pthread - no
-- Found Threads: TRUE  
-- Found FFmpeg: /usr/aarch64-linux-gnu/usr/lib/../lib64/libavformat.so (found version "58.45.100") found components: avformat avutil swscale swresample avcodec 
-- XCB[XINPUT]: Found component XINPUT
-- Found XCB: /usr/aarch64-linux-gnu/usr/lib/libxcb-xinput.so  found components: XINPUT 
-- XCB[XCB]: Found component XCB
-- XCB[COMPOSITE]: Found component COMPOSITE
-- XCB[DAMAGE]: Found component DAMAGE
-- XCB[DRI2]: Found component DRI2
-- XCB[GLX]: Found component GLX
-- XCB[RANDR]: Found component RANDR
-- XCB[RENDER]: Found component RENDER
-- XCB[SHAPE]: Found component SHAPE
-- XCB[SHM]: Found component SHM
-- XCB[SYNC]: Found component SYNC
-- XCB[XFIXES]: Found component XFIXES
-- XCB[XTEST]: Found component XTEST
-- XCB[XV]: Found component XV
-- XCB[XINPUT]: Found component XINPUT
-- XCB[XINERAMA]: Found component XINERAMA
-- Found XCB: /usr/aarch64-linux-gnu/usr/lib/libxcb-xinput.so;/usr/aarch64-linux-gnu/usr/lib/libxcb.so;/usr/aarch64-linux-gnu/usr/lib/libxcb-composite.so;/usr/aarch64-linux-gnu/usr/lib/libxcb-damage.so;/usr/aarch64-linux-gnu/usr/lib/libxcb-dri2.so;/usr/aarch64-linux-gnu/usr/lib/libxcb-glx.so;/usr/aarch64-linux-gnu/usr/lib/libxcb-randr.so;/usr/aarch64-linux-gnu/usr/lib/libxcb-render.so;/usr/aarch64-linux-gnu/usr/lib/libxcb-shape.so;/usr/aarch64-linux-gnu/usr/lib/libxcb-shm.so;/usr/aarch64-linux-gnu/usr/lib/libxcb-sync.so;/usr/aarch64-linux-gnu/usr/lib/libxcb-xfixes.so;/usr/aarch64-linux-gnu/usr/lib/libxcb-xtest.so;/usr/aarch64-linux-gnu/usr/lib/libxcb-xv.so;/usr/aarch64-linux-gnu/usr/lib/libxcb-xinput.so;/usr/aarch64-linux-gnu/usr/lib/libxcb-xinerama.so   
-- Found PulseAudio - Audio Monitor enabled
-- Checking for modules 'gio-2.0;gio-unix-2.0'
--   Found gio-2.0, version 2.68.2
--   Found gio-unix-2.0, version 2.68.2
-- Using libavcodec for image loading in libobs
-- Found ZLIB: /usr/aarch64-linux-gnu/usr/lib/../lib64/libz.so  
-- Found Wayland: /usr/aarch64-linux-gnu/usr/lib/libwayland-client.so;/usr/aarch64-linux-gnu/usr/lib/libwayland-server.so;/usr/aarch64-linux-gnu/usr/lib/libwayland-egl.so;/usr/aarch64-linux-gnu/usr/lib/libwayland-cursor.so  found components: Client 
-- XCB[XCB]: Found component XCB
-- XCB[RANDR]: Found component RANDR
-- XCB[SHM]: Found component SHM
-- XCB[XFIXES]: Found component XFIXES
-- XCB[XINERAMA]: Found component XINERAMA
-- Found XCB: /usr/aarch64-linux-gnu/usr/lib/libxcb.so;/usr/aarch64-linux-gnu/usr/lib/libxcb-randr.so;/usr/aarch64-linux-gnu/usr/lib/libxcb-shm.so;/usr/aarch64-linux-gnu/usr/lib/libxcb-xfixes.so;/usr/aarch64-linux-gnu/usr/lib/libxcb-xinerama.so  found components: XCB RANDR SHM XFIXES XINERAMA 
-- Found Libv4l2: /usr/aarch64-linux-gnu/usr/lib/libv4l2.so  
-- Found UDev: /usr/aarch64-linux-gnu/usr/lib/libudev.so  
-- Checking for module 'jack'
--   Found jack, version 1.9.18
-- Found jack: /usr/aarch64-linux-gnu/usr/lib/libjack.so
-- Found ALSA: /usr/aarch64-linux-gnu/usr/lib/libasound.so (found version "1.2.4") 
-- Found LibVLC: /usr/aarch64-linux-gnu/usr/include/vlc  
-- Could NOT find Sndio (missing: Sndio_LIBRARY Sndio_INCLUDE_DIR) 
-- Sndio not found, disabling Sndio plugin
-- Found Libx264: /usr/aarch64-linux-gnu/usr/lib/../lib64/libx264.so  
-- Found FFmpeg: /usr/aarch64-linux-gnu/usr/lib/../lib64/libavcodec.so (found version "58.91.100") found components: avcodec avfilter avdevice avutil swscale avformat swresample 
-- Found FFmpeg: /usr/aarch64-linux-gnu/usr/lib/../lib64/libavcodec.so (found version "58.91.100") found components: avcodec avutil avformat 
-- Found MbedTLS: /usr/aarch64-linux-gnu/usr/lib/../lib64/libmbedtls.so;/usr/aarch64-linux-gnu/usr/lib/../lib64/libmbedcrypto.so;/usr/aarch64-linux-gnu/usr/lib/../lib64/libmbedx509.so  
-- Checking for module 'libftl'
--   No package 'libftl' found
-- Found Libspeexdsp: /usr/aarch64-linux-gnu/usr/lib/../lib64/libspeexdsp.so  
-- Configuring incomplete, errors occurred!
See also "/builddir/obs-studio-27.0.1/build/CMakeFiles/CMakeOutput.log".
See also "/builddir/obs-studio-27.0.1/build/CMakeFiles/CMakeError.log".
```

### Steps to Reproduce

1. cross compile from x86_64 for aarch64 (with glibc)

### Anything else we should know?

this issue happens here: https://github.com/obsproject/obs-studio/blob/dcd491541e5e6f68f57f7ab718badac2c01eb7c5/cmake/Modules/FindThreads.cmake#L87-L94

**Comment by RytoEX** [comment]
Might be fixed by #4331.

**Comment by paper42** [comment]
> Might be fixed by #4331.

I can confirm that the PR fixes this bug
